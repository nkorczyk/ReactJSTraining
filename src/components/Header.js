import React from 'react';
import Title from './Title';

const Header = () => {
  return (
    <div className="wrapper">
      <Title />
      <h4 className="white-text text-darken-2">FIND YOUR MOVIE</h4>
      <input type="text" placeholder="Enter the title" className="input-field white-text bcg"></input>
      <div>
        <h6 className="white-text text-darken-2 left">SEARCH BY</h6>
        <button className="red lighten-1 btn button1">TITLE</button>
        <button className="grey darken-2 btn button1">GENRE</button>
        <button className="red lighten-1 btn right">SEARCH</button>
      </div>

    </div>
  )
}

export default Header;
