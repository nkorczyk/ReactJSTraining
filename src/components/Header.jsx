import React, { Component } from 'react';
import Title from './Title';
import CONSTANTS from '../constants/constants';
import { loadMovies, searchBy, searchMovieChange } from '../actions/actionCreator';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

class Header extends Component {

  handleSearchByClick = (event) => {
    this.props.searchBy(event.target.id);
  }

  handleChange = (event) => {
    this.props.searchMovieChange(event.target.value);
  }

  render() {
    return (
      <header className="header">
        <Title />
        <h4 className="white-text text-darken-2">{CONSTANTS.FIND}</h4>
        <input type="text" placeholder="Enter the title" className="input-field white-text"
          onChange={this.handleChange} />
        <div>
          <h6 className="white-text text-darken-2 left">{CONSTANTS.SEARCH_BY}</h6>
          <button id="title" className="lighten-1 btn buttons"
            onClick={this.handleSearchByClick}>{CONSTANTS.TITLE}</button>
          <button id="genre" className="lighten-1 btn buttons"
            onClick={this.handleSearchByClick}>{CONSTANTS.GENRE}</button>
          <button id="search" className="red lighten-1 btn right"
            onClick={this.props.loadMovies}>{CONSTANTS.SEARCH}</button>
        </div>
      </header>
    )
  }
}

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    loadMovies,
    searchMovieChange,
    searchBy,
  }, dispatch)
};

export { Header };
export default connect(null, mapDispatchToProps)(Header);
