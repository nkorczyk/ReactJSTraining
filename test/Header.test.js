import React from 'react';
import { mount } from 'enzyme';
import Header from "../src/components/Header";
import configureMockStore from 'redux-mock-store';

const initialState = {
  search: {
    searchby: 'TITLE',
    phrase: ''
  }
};

const mockStore = configureMockStore();

describe('Header', () => {
  const store = mockStore(initialState);
  it('should render Header component', () => {
    const component = mount(
      <Header searchby={'TITLE'}
        handleSearchByClick={jest.fn()}
        store={store} />);
    expect(component).toMatchSnapshot();
  });
});
