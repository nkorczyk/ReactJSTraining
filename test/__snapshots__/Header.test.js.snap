// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header component 1`] = `
<Connect(Header)
  handleSearchByClick={[MockFunction]}
  searchby="TITLE"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Header
    handleSearchByClick={[MockFunction]}
    loadMovies={[Function]}
    searchBy={[Function]}
    searchMovieChange={[Function]}
    searchby="TITLE"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <header
      className="header"
    >
      <Title>
        <div>
          <h5
            className="red-text text-darken-2"
          >
            netflixroulette
          </h5>
        </div>
      </Title>
      <h4
        className="white-text text-darken-2"
      >
        FIND YOUR MOVIE
      </h4>
      <input
        className="input-field white-text"
        onChange={[Function]}
        placeholder="Enter the title"
        type="text"
      />
      <div>
        <h6
          className="white-text text-darken-2 left"
        >
          SEARCH BY
        </h6>
        <button
          className="lighten-1 btn buttons"
          id="title"
          onClick={[Function]}
        >
          TITLE
        </button>
        <button
          className="lighten-1 btn buttons"
          id="genre"
          onClick={[Function]}
        >
          GENRE
        </button>
        <button
          className="red lighten-1 btn right"
          id="search"
          onClick={[Function]}
        >
          SEARCH
        </button>
      </div>
    </header>
  </Header>
</Connect(Header)>
`;
